src/drm.c

```diff
@@ -534,18 +534,14 @@ int modeset_prepare(int fd, struct modeset_output *output_list, uint16_t mode_wi
 
 		out = modeset_output_create(fd, res, conn, mode_width, mode_height, mode_vrefresh);
 		drmModeFreeConnector(conn);
-		if (!out)
-			continue;
-
-		*output_list = *out;
```

Comment:
here we used to re-assign already alocated `output_list` with pointer alocated in `modeset_output_create` creating a memleak. Not anymore!